### PR TITLE
(GH-448) Add extra metadata to nuspec files

### LINKED ISSUE #448: Add extra metadata to nuspec files
There's a few warnings coming through from Chocolatey's validation steps about missing optional metadata in the Boxstarter packages. eg.

> - Package contains dependencies with no specified version. You should at least specify a minimum version of a dependency. More...
> - The nuspec has been enhanced to allow packageSourceUrl, pointing to the url where the package source resides. This is a strong guideline because it simplifies collaboration. Please add it to the nuspec. More...
> - Summary (summary) is a short explanation of the software. Please include summary in the nuspec

and

> The nuspec has been enhanced to allow more information related to the software. More... Please consider adding one or more of the following to the nuspec, if available:
> 
> - docsUrl - points to the location of the wiki or docs of the software
> - mailingListUrl - points to the forum or email list group for the software
> - bugTrackerUrl - points to the location where issues and tickets can be accessed
> - projectSourceUrl - points to the location of the underlying software source
>

### PR DESCRIPTION
**Pull request by flcdrg**
## Description

Add missing metadata to nuspec files so there are less validation warnings when packages are submitted to Chocolatey public repo.

## Related Issue

Resolves #448

## Motivation and Context
Nicer Chocolatey packages

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/chocolatey/boxstarter/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: BuildScripts/default.ps1 [hunk 1]
```diff
@@ -317,7 +317,7 @@ function PackDirectory($path, [switch]$AddReleaseNotes){
                    $nuspec.package.metadata.ReplaceChild($newReleaseNotes, $oldReleaseNotes) | Out-Null
                    $nuspec.Save($_)
                  }
-                 .$nugetExe pack $_ -OutputDirectory $path -NoPackageAnalysis -version $version
+                 choco pack $_ --OutputDirectory $path --version $version
```
Comment (pauby) [inline BuildScripts/default.ps1:320]: Can we remove this change for two reasons:

1. It doesn't fir in with the PR title / scope;
2. We are going to look at changing the build in future and would ratehr do all of that, including this type of change if it's needed, in there;

Comment (flcdrg) [reply BuildScripts/default.ps1:320]: The changes won't work without it. `nuget.exe` doesn't recognise the Chocolatey metadata that is being added and throws an error.

**Comment by pauby** [comment]
@flcdrg As we're going to be looking at #465 it might be better to leave this here for now and rebase it or close it and look at it as part of the larger piece of work rather than in isolation? Thoughts?

**Comment by flcdrg** [comment]
The majority of this change is to improve the nuspec files. There's just one line that updates the build script. I've just rebased it now for good measure.

File: BuildScripts/default.ps1 (hunk 1 above)
Comment (gep13) [reply BuildScripts/default.ps1:310]: Agreed, this change is required, and fits in with the other changes that are going into the build.

**Review by gep13 — APPROVED** [review]
LGTM!

**Comment by gep13** [comment]
@flcdrg thanks for getting all those updated!

**Pull request merged**